compendium/CM/include/cppmicroservices/cm/ConfigurationAdmin.hpp

```diff
@@ -0,0 +1,95 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+
+#ifndef ConfigurationAdmin_hpp
```

Comment:
Just wondering if ConfigurationAdmin_hpp is enough of a guard. Should include the component name to be sure?
CppMicroServices_ConfigurationAdmin.

Reply:
In this case, it is fine. This file is part of the public interface of ConfigAdmin and none of the other interface header files use this header guard as they utilize their file names as the include guard. Additionally, none of the private headers in the implementation of ConfigAdmin use the same include guard so in this case we are okay. I believe the convention for this project is that the include guard should be the file name; there may be some special cases that need to be considered with other files but this isn't one of them from what I can tell.